modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionVulcanBatchEngineTaskItemDelegate.java

```diff
@@ -0,0 +1,265 @@
+	}
+
+	@Override
+	public void setContextBatchUnsafeConsumer(
+		UnsafeBiConsumer
+			<Collection<ObjectDefinition>,
+			 UnsafeConsumer<ObjectDefinition, Exception>, Exception>
+				contextBatchUnsafeConsumer) {
+
+		throw new UnsupportedOperationException();
+	}
+
+	@Override
+	public void setContextCompany(Company contextCompany) {
+		if (_log.isWarnEnabled()) {
+			_log.warn(
+				"Provided company is ignored in BATCH. Current user's " +
+					"company would be used instead.");
+		}
+	}
+
+	@Override
+	public void setContextUser(User contextUser) {
+		_contextUser = contextUser;
+	}
+
+	@Override
+	public void setLanguageId(String languageId) {
+		_contextAcceptLanguage = new AcceptLanguage() {
+
+			@Override
+			public List<Locale> getLocales() {
+				return null;
+			}
+
+			@Override
+			public String getPreferredLanguageId() {
+				return languageId;
+			}
+
+			@Override
+			public Locale getPreferredLocale() {
+				return LocaleUtil.fromLanguageId(languageId);
+			}
+
+		};
+	}
+
+	@Override
+	public void update(
+			Collection<ObjectDefinition> items,
+			Map<String, Serializable> parameters)
+		throws Exception {
+
+		throw new UnsupportedOperationException();
+	}
+
+	protected <T, R, E extends Throwable> List<R> transformToList(
+		T[] array, UnsafeFunction<T, R, E> unsafeFunction) {
+
+		return TransformUtil.transformToList(array, unsafeFunction);
+	}
+
+	private com.liferay.object.model.ObjectDefinition _create(
+			ObjectDefinition objectDefinition)
+		throws Exception {
+
+		return _objectDefinitionService.addCustomObjectDefinition(
+			LocalizedMapUtil.getLocalizedMap(objectDefinition.getLabel()),
+			objectDefinition.getName(), objectDefinition.getPanelAppOrder(),
+			objectDefinition.getPanelCategoryKey(),
+			LocalizedMapUtil.getLocalizedMap(objectDefinition.getPluralLabel()),
+			objectDefinition.getScope(), objectDefinition.getStorageType(),
+			transformToList(
+				objectDefinition.getObjectFields(),
+				objectField -> ObjectFieldUtil.toObjectField(
+					objectField, _objectFieldLocalService,
+					_objectFieldSettingLocalService,
+					_objectFilterLocalService)));
+	}
+
+	private void _createWithWorkflow(ObjectDefinition objectDefinition)
+		throws Exception {
+
+		com.liferay.object.model.ObjectDefinition objectDefinition1 = _create(
+			objectDefinition);
+
+		Status status = objectDefinition.getStatus();
+
+		if (status.getCode() == WorkflowConstants.STATUS_APPROVED) {
+			_objectDefinitionService.publishCustomObjectDefinition(
+				objectDefinition1.getObjectDefinitionId());
+		}
+	}
+
+	private UnsafeConsumer<ObjectDefinition, Exception> _getUnsafeConsumer(
+		Map<String, Serializable> parameters) {
+
+		Boolean applyWorkflowStatus = (Boolean)parameters.getOrDefault(
+			"applyWorkflowStatus", Boolean.TRUE);
+
+		String createStrategy = (String)parameters.getOrDefault(
+			"createStrategy", "INSERT");
+
+		if ("INSERT".equalsIgnoreCase(createStrategy)) {
+			if (!applyWorkflowStatus) {
```

Comment:
I think if we discuss the necessity of including the applyWorkflowStatus as a query parameter in the POST method of the REST API with PM, we could reuse the current batch default implementations and I also think that the performance would be pretty similar, so I would not see the need of duplicating the VulcanBatchEngineTaskItemDelegate implementation

Reply:
Open API may be updated with `applyWorkflowStatus`query parameter but it should be documented in description that it is used for internal communication between batch-engine and *ResourceImpl.

Reply:
But maybe Pablo would accept to provide also that functionality to the normal user